### PR TITLE
ReadRange planner operation and ReadFilter storage call

### PR DESCRIPTION
**Pull request by jlapacik**
Closes #

_Briefly describe your proposed changes:_

This PR adds a new physical operation to the planner called ReadRange. It is represented by the `ReadRangePhysSpec`. This new physical operation is constructed from the pattern `from |> range`. In essence it is the result of pushing a 'range' into a 'from'.

This new planner operation maps to a new ReadFilter storage call which performs a simple read with an optional predicate.

  - [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [ ] Rebased/mergeable
  - [ ] Tests pass
  - [ ] http/swagger.yml updated (if modified Go structs or API)
  - [ ] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: query/stdlib/influxdata/influxdb/operators.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"fmt"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/values"
+)
+
+const ReadRangePhysKind = "ReadRangePhysKind"
+
+type ReadRangePhysSpec struct {
```
Comment (wolffcm) [inline query/stdlib/influxdata/influxdb/operators.go]: This is a divergence from how we've named the other `ProcedureSpec`s.  I think I like it though.  If there is a 1:1 correspondence with physical specs and RPCs, maybe it should match the RPC name as close as possible?

File: query/stdlib/influxdata/influxdb/source.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/plan"
+	platform "github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/query"
+	"github.com/influxdata/influxdb/tsdb/cursors"
+)
+
```
Comment (wolffcm) [inline query/stdlib/influxdata/influxdb/source.go]: So everything in this file will be the common parts of the various InfluxDB read operations, is that right?

File: query/stdlib/influxdata/influxdb/operators.go (hunk 1 above)
Comment (jlapacik) [reply query/stdlib/influxdata/influxdb/operators.go]: Perhaps change `ReadRangePhysSpec` to just `ReadRange`? I wouldn't mind that.

File: query/stdlib/influxdata/influxdb/source.go (hunk 2 above)
Comment (jlapacik) [reply query/stdlib/influxdata/influxdb/source.go]: Correct. We will need a `ReadGroupSource` and a source for meta queries which will be placed here.

**Comment by jlapacik** [comment]
@wolffcm yes there will be a 1:1 mapping between physical read operations and RPC calls with one caveat. The planner makes a distinction between ReadRange and ReadFilter, but storage does not. That is, both ReadRange and ReadFilter will map to the same RPC call as there is no difference to how they are answered at storage.

This simplifies the filter push down rule slightly, since a filter will only be able to be pushed down into a ReadRange operation. In the case that a "non-pushable" filter remains after applying the rule, no attempt will be made to push it down to storage.

File: storage/reads/datatypes/storage_common.proto [hunk 3]
```diff
@@ -16,6 +16,9 @@ service Storage {
   // Read performs a read operation using the given ReadRequest
   rpc Read (ReadRequest) returns (stream ReadResponse);
 
+  // ReadFilter performs a filter operation at storage
+  rpc ReadFilter (ReadFilterRequest) returns (stream ReadResponse);
```
Comment (stuartcarnie) [inline storage/reads/datatypes/storage_common.proto]: Presumably you are dropping `Range` (as in `ReadRangeFilter`) given a time range is assumed for all requests and therefore redundant

Comment (jlapacik) [reply storage/reads/datatypes/storage_common.proto]: Yes there's no reason to have a separate ReadRange RPC as everything is a filter from the perspective of storage. The planner will ensure a range is always included in a call to storage or else planning will fail.

**Pull request merged**